mat64/pool_test.go

```diff
@@ -15,26 +16,29 @@ import (
 func (s *S) TestPool(c *check.C) {
 	for i := 1; i < 10; i++ {
 		for j := 1; j < 10; j++ {
-			var w, m *Dense
+			m := NewDense(i, j, nil)
 			for k := 0; k < 5; k++ {
-				w = getWorkspace(i, j, true)
-				m = NewDense(i, j, nil)
-				c.Check(w.mat, check.DeepEquals, m.mat)
-				c.Check(w.capRows, check.DeepEquals, m.capRows)
-				c.Check(w.capCols, check.DeepEquals, m.capCols)
-				c.Check(cap(w.mat.Data) < 2*len(w.mat.Data), check.Equals, true, check.Commentf("r: %d c: %d -> len: %d cap: %d", i, j, len(w.mat.Data), cap(w.mat.Data)))
-			}
-			w.Set(0, 0, math.NaN())
-			for k := 0; k < 5; k++ {
-				putWorkspace(w)
-			}
-			for k := 0; k < 5; k++ {
-				w = getWorkspace(i, j, true)
-				m = NewDense(i, j, nil)
-				c.Check(w.mat, check.DeepEquals, m.mat)
-				c.Check(w.capRows, check.DeepEquals, m.capRows)
-				c.Check(w.capCols, check.DeepEquals, m.capCols)
-				c.Check(cap(w.mat.Data) < 2*len(w.mat.Data), check.Equals, true, check.Commentf("r: %d c: %d -> len: %d cap: %d", i, j, len(w.mat.Data), cap(w.mat.Data)))
+				work := make([]*Dense, rand.Intn(10))
```

Comment:
Why is this a random number? It seems at least it should be Intn() + 1 so that at last one matrix is created.